Development setup in readme, and some adjustments

jancborchardt opened the pull request:
As mentioned in https://github.com/nextcloud/server/pull/12186#issuecomment-435017943, adding some info regarding frontend development setup to the main readme.

Also fix some structure of readme, adjust wording, and add more emojis 🤩

Please review @skjnldsv @juliushaertl @camilasan @rullzer @danxuliu @MariusBluem

jancborchardt committed:
Add front-end building documentation to readme

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Readme: adjust structure, wording, small fixes

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Readme: Fix screenshot link

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

juliushaertl commented on README.md:
```diff
@@ -9,43 +9,100 @@
 2. [Pick a good first issue](https://github.com/nextcloud/server/labels/good%20first%20issue) :notebook:
-3. Create a branch, a [Pull Request](https://opensource.guide/how-to-contribute/#opening-a-pull-request) and `@mention` the people from the issue :computer:
-4. Wait for it to get merged and :tada:
+3. Create a branch, a [pull request](https://opensource.guide/how-to-contribute/#opening-a-pull-request) and `@mention` the people from the issue to review :computer:
```
Can we add a note about adding the sign off message here? This is not obvious from the readme and missing in a lot of first time contributors PRs.

jancborchardt replied:
Added! Is it good like that?

jancborchardt committed:
Add sentence about non-technical contributions

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Readme: Add note about signing off commits

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt commented:
Also added sentence about non-technical contributions with link to https://nextcloud.com/contribute/, and fixed the screenshot! (@jospoortvliet @MariusBluem if the structure in the Screenshots repo is changed, links will break. ;) We should keep the naming.)

jancborchardt committed:
Readme: Replace Github emoji codes with real emoji

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

rullzer reviewed (approved):
Nice :)

splitt3r commented on README.md:
```diff
@@ -7,45 +7,104 @@
+
+[![BrowserStack](https://user-images.githubusercontent.com/45821/41675934-61fa3442-74c4-11e8-8c8e-90768c56ba08.png)](https://www.browserstack.com/)
+[![WAVE](https://wave.webaim.org/img/wavelogo.png)](https://wave.webaim.org/extension/)
+[![](https://developers.google.com/web/tools/lighthouse/images/lighthouse-icon-128.png) Lighthouse](https://developers.google.com/web/tools/lighthouse/)
```
```suggestion
[![Lighthouse](https://developers.google.com/web/tools/lighthouse/images/lighthouse-icon-128.png)](https://developers.google.com/web/tools/lighthouse/)
```

splitt3r replied:
Personally i think there are to many smileys now :grin:. Anyway i think the logos under "Tools we use" should be a bit smaller or even same size.

jancborchardt replied:
@splitt3r I see why you suggest that. :) This is intentional though as the logo I found has no text. And the others do have text in the logo. So for Lighthouse it is indeed text, and not just alternative text for the image.

jancborchardt replied:
> Anyway i think the logos under "Tools we use" should be a bit smaller or even same size.

Good point! Switched to using HTML in that section and reduced the size.

jancborchardt replied:
Actually didn’t really work out with the HTML – just switched to text-only. :)

juliushaertl replied:
I'm not sure if browserstack required to include the logo for the free accounts, but @rullzer probably knows.

jancborchardt committed:
Readme: Add line breaks between tool images

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Readme: Reduce size of Tools logos, use HTML for part

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>

jancborchardt committed:
Readme: Simply use emojis instead of images for tools section

Signed-off-by: Jan-Christoph Borchardt <hey@jancborchardt.net>
